test/unit/connectionTest.ts

```diff
@@ -282,6 +282,24 @@ describe('Connection', () => {
     expect(toolingQuerySpy.firstCall.args[1]).to.have.property('maxFetch', 50000);
   });
 
+  it('tooling.autoFetchQuery() should not throw a warning when records is empty', async () => {
+    const soql = 'TEST_SOQL';
+
+    const queryResponse = { totalSize: 5, done: true, records: [] };
+    requestMock.returns(Promise.resolve(queryResponse));
```

Comment:
is there a reason you can't do `requestMock.resolves(queryResponse)`?

Reply:
not that I know of.  I kept it consistent with the rest of the tests. 

gonna change them all.